tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,327 @@
+    program_pid = -1
+
+    program_list = duthost.shell("docker exec {} supervisorctl".format(container_name))
+    for program_info in program_list["stdout_lines"]:
+        if program_info.find(critical_group) != -1:
+            program_name = program_info.split()[0].split(':')[1].strip(' \n')
+            program_status = program_info.split()[1].strip()
+            if program_status in ["EXITED", "STOPPED", "STARTING"]:
+                program_pid = -1
+            else:
+                program_pid = program_info.split()[3].strip(' ,')
+
+            group_program_info[program_name].append(program_status)
+            group_program_info[program_name].append(program_pid)
+            
+    return group_program_info
+
+def get_program_info(duthost, container_name, program_name):
+    """
+    @summary: Get program running status and its pid by analyzing the command
+              output of "docker exec <container_name> supervisorctl"
+    @return:  Program running status and its pid
+    """
+    program_status = None
+    program_pid = -1
+
+    program_list = duthost.shell("docker exec {} supervisorctl".format(container_name))
+    for program_info in program_list["stdout_lines"]:
+        if program_info.find(program_name) != -1:
+            program_status = program_info.split()[1].strip()
+            if program_status == "RUNNING":
+                program_pid = int(program_info.split()[3].strip(' ,'))
+            break
+
+    if program_status:
+        logger.info("Found program {} in the {} state with pid {}".format(program_name, program_status, program_pid))
+            
+    return program_status, program_pid
+
+def get_autorestart_container_and_state(duthost):
+    """
+    @summary: Get container names and its autorestart states by analyzing 
+              the command output of "show container feature autorestart"
+    @return:  A dictionary where keys are the names of containers which have the 
+              autorestart feature implemented and values are the autorestart feature
+              state for that container
+    """
+    autorestart_containers = {}
+
+    show_cmd_output = duthost.shell(CMD_CONTAINER_FEATURE_AUTORESTART)
+    for line in show_cmd_output["stdout_lines"]:
+        container_name = line.split()[0].strip()
+        container_state = line.split()[1].strip(' \n')
+        if container_state in ["enabled", "disabled"]:
+           autorestart_containers[container_name] = container_state
+
+    return autorestart_containers
+
+def get_disabled_container_list(duthost):
+    """
+    @summary: Get the container/service names which are disabled by default
+    @return: A list includes the names of disabled container/service
+    """
+    disabled_containers = []
+    container_list = []
+
+    redis_cli_keys_output = duthost.shell("redis-cli -n 4 keys 'FEATURE|*'")
+    for line in redis_cli_keys_output["stdout_lines"]:
+        container_list.append(line.strip(' \n').split('|')[1])
```

Comment:
Is there a reason to define the chars to strip as `' \n'`? By default, `strip()` will strip all leading and trailing whitespace.